src/docs/eviction.dox

```diff
@@ -0,0 +1,103 @@
+/*! @m_page{{c,java},eviction, Eviction}
+
+All the operations performed in WiredTiger are on the data read into
+preconfigured amount of memory. WiredTiger uses memory as a cache for
+all the data on the disk and the data in memory forms the current
+working set.
+
+Since memory is limited and can't hold all the data stored on the disk,
+WiredTiger has to continuously move data that is currently not being
+accessed out of the memory to free up enough space to read data that
+are requested by the user but currently reside on the disk back into
+memory. This process is called eviction.
+
+@section overview Eviction overview
+
+The WiredTiger eviction runs in the background with one eviction server
+thread and several eviction worker threads.
+
+1) The eviction server thread walks the btrees and put the pages into
+the eviction queue.
+2) The eviction workers continuously remove pages from the eviction
+queue and try to evict them.
+
+Eviction has to obtain exclusive access to the page as there may be
+application threads still reading content on the page that is being
+evicted.
+
+There are three types of eviction in WiredTiger:
+
+1) Clean eviction
+2) History store eviction
+3) Update restore eviction
+
+@section clean_eviction Clean eviction
+
+In the case of clean eviction, there is no dirty content on the page
+and it is directly removed from the memory. After that, the page
+remains solely on the disk unchanged.
+
+@section hs_eviction History store eviction
+
+For history store eviction, the page is modified but the changes are
+all committed. Therefore, the page should be clean and can be removed
+from memory after the dirty changes are written to the new disk image
+in the process called reconciliation. WiredTiger writes the newest
+committed value of each key to the disk image. All the older versions
+are moved to the history store table. The new disk image is
+subsequently flushed to the data file on the disk and the page is
+removed from memory.
+
+The history store table is the internal table for WiredTiger to store
+historical versions for all the user-created tables. The pages in the
+history store table need to be evicted and loaded back to memory as
+well. Each entry in the history store represents a historical value
+of a key in the user-created tables. Different from the user-created
+tables, all the content on the history store page is committed as the
+historical values it stores must all be committed, and each entry only
+has one value and a deletion marking the expiration of the entry on
+its update chain. Therefore, history store pages should always be clean
+after reconciliation and can be evicted from memory. Besides, it
```

Comment:
I'm very confused by the last sentence. When you say "it prevents eviction", what is meant by "it"? Reconciliation? I'm assuming that we are preventing something bad from happening but I don't understand what exactly it is or how it happens. Could you maybe expand on this sentence to clarify what you are trying to say?

Reply:
Reworded